HW11/Selenium.cs

```diff
@@ -0,0 +1,463 @@
+            Assert.IsTrue(selectedYes.Selected);
+
+            var selectedYesResult = _driver.FindElement(By.CssSelector("[class='mt-3']")).Text;
+            Assert.AreEqual("You have selected Yes", selectedYesResult);
+
+            impressiveRadioButton.Click();
+            var selectedImpresive = _driver.FindElement(By.XPath("//*[@id=\'impressiveRadio\']"));
+            Assert.IsTrue(selectedImpresive.Selected);
+            Assert.IsFalse(selectedYes.Selected);
+
+            var selectedImpresiveResult = _driver.FindElement(By.CssSelector("[class='mt-3']")).Text;
+            Assert.AreEqual("You have selected Impressive", selectedImpresiveResult);
+        }
+
+        [Test]
+        public void WebTables()
+        {
+            _driver.Navigate().GoToUrl("https://demoqa.com/webtables");
+            var table = _driver.FindElement(By.XPath("//div[@class='ReactTable -striped -highlight']"));
+            var searchBox = _driver.FindElement(By.Id("searchBox"));
+            var addButton = _driver.FindElement(By.Id("addNewRecordButton"));
+
+            Assert.IsTrue(table.Displayed); //expected result of 1 
+            Assert.IsTrue(searchBox.Displayed); //expected result of 1 
+            Assert.IsTrue(addButton.Displayed); //expected result of 1 
+
+            addButton.Click(); // step 2
+
+            var regForm = _driver.FindElement(By.Id("registration-form-modal"));
+
+            _driverWait.Until(drv =>
+                {
+                    if (drv.FindElements(By.Id("registration-form-modal")).Count > 0)
+                    {
+                        return true;
+                    }
+                    return false;
+                });
+
+            Assert.IsTrue(regForm.Displayed); // exp result 2
+
+            var firstNameField = _driver.FindElement(By.Id("firstName"));
+            var lastNameField = _driver.FindElement(By.Id("lastName"));
+            var emailField = _driver.FindElement(By.Id("userEmail"));
+            var ageField = _driver.FindElement(By.Id("age"));
+            var salaryField = _driver.FindElement(By.Id("salary"));
+            var departmentField = _driver.FindElement(By.Id("department"));
+
+            Assert.IsTrue(firstNameField.Displayed); // exp result 2
+            Assert.IsTrue(lastNameField.Displayed); // exp result 2
+            Assert.IsTrue(emailField.Displayed); // exp result 2
+            Assert.IsTrue(ageField.Displayed); // exp result 2
+            Assert.IsTrue(salaryField.Displayed); // exp result 2
+            Assert.IsTrue(departmentField.Displayed); // exp result 2
+
+            var submitButton = _driver.FindElement(By.Id("submit"));
+            submitButton.Click(); // step 2
+
+            IWebElement firstNameFieldError = _driver.FindElement(By.XPath("//input[@id='firstName']"));
+
+            _driverWait.Until(drv =>
+            {
+                if (firstNameFieldError.GetCssValue("border-color").Contains("rgb(220, 53, 69)"))
```

Comment:
no need for if, just set `firstNameFieldError.GetCssValue("border-color").Contains("rgb(220, 53, 69)")`. Your wait does not work properly, It waits for both condition and will never fail.
By default webdriverwait waits for true result